BDArmory/FX/ExplosionFX.cs

```diff
@@ -524,8 +547,51 @@ private bool IsInLineOfSight(Part part, Part explosivePart, out RaycastHit hit,
                     if (intermediateParts)
                     {
                         var partHP = partHit.Damage();
-                        if (ProjectileUtils.IsArmorPart(partHit)) partHP = 100;
+                        if (ProjectileUtils.IsArmorPart(partHit)) partHP = BDArmorySettings.EXP_PEN_RESIST_MULT * 100;
                         var partArmour = partHit.GetArmorThickness();
+                        var Armor = part.FindModuleImplementing<HitpointTracker>();
```

Comment:
You want `partHit` (the intermediate part) here, not `part`.

Also, I see that the next thing I need to optimise is access to part modules so that we can avoid `FindModuleImplementing<>()` calls.

Reply:
FindModule<HitpointTracker>() is going to be a big offender here, given there isn't really a better way of accessing it (yet), but is needed by so many things. A ``List<PartModule> hitpointTrackers;`` for all parts on a vessel and use VesselModuleRegistry, perhaps?